src/spectral.ts

```diff
@@ -240,6 +243,20 @@ export class Spectral {
 
     return resolveOpts;
   };
+
+  private static _generateUnrecognizedFormatError(parsedResult: IParsedResult): IRuleResult {
+    return {
+      range: parsedResult.getLocationForJsonPath(parsedResult.parsed, [], true)?.range || {
+        start: { character: 0, line: 0 },
+        end: { character: 0, line: 0 },
+      },
+      message: 'Provided file format does not match any of the registered formats',
+      code: 'unrecognized-format',
+      severity: DiagnosticSeverity.Information,
```

Comment:
Info or error? E.g. should probably exit with a 0 if this occurs, so that tests fail? For example if user removes the `swagger` property from their openapi file, it is definitely invalid, but Spectral will not identify it as openapi at all and thus run no rules.

Reply:
You can make still make Spectral fail if you use `fail-severity-level` flag.
In any case, I agree that warning feels like a more appropriate severity. Will change.

Reply:
Yeah let’s make it a warning. The CLI is already set to panic about this so no worries there.